src/java.base/share/classes/jdk/internal/classfile/impl/AbstractDirectBuilder.java

```diff
@@ -52,6 +52,8 @@ public void setOriginal(M original) {
     }
 
     public void writeAttribute(Attribute<?> a) {
-        attributes.withAttribute(a);
+        if (Util.isAttributeAllowed(a, context.attributesProcessingOption())) {
+            attributes.withAttribute(a);
+        }
     }
```

Comment:
The "is allowed" check should always say yes for unbound attributes, and only apply the option for bound attributes.  (You may already have done that, but I wanted to make sure this was captured in the review.)